http.go

```diff
@@ -231,6 +232,15 @@ func GetCustomHTTPRetryableClient(retry HTTPRetryFunction, delay HTTPDelayFuncti
 	return client
 }
 
+// GetCustomHTTPRetryableClientWithTransport allows you to generate an HTTP client
+// that automatically retries failed request based on the provided
+// custom logic and takes in a github transport.
+func GetCustomHTTPRetryableClientWithGithubTransport(itr *ghinstallation.Transport, retry HTTPRetryFunction, delay HTTPDelayFunction) *http.Client {
```

Comment:
If we want to make this a little more generalizable, we could change this function [to accept any `http.RoundTripper` implementation and set it as the Transport](https://pkg.go.dev/net/http@go1.20.3#Client). That way, we're not limited to just `ghinstallation.Transport` since we can pass in anything that satisfies the interface.

Reply:
that makes more sense